lldb/tools/lldb-dap/Handler/InitializeRequestHandler.cpp

```diff
@@ -167,14 +167,23 @@ static void EventThreadFunction(DAP &dap) {
             // stop events which we do not want to send an event for. We will
             // manually send a stopped event in request_configurationDone(...)
             // so don't send any before then.
-            if (dap.configuration_done_sent) {
-              // Only report a stopped event if the process was not
-              // automatically restarted.
-              if (!lldb::SBProcess::GetRestartedFromEvent(event)) {
-                SendStdOutStdErr(dap, process);
-                SendThreadStoppedEvent(dap);
+            {
```

Comment:
What if we didn't start the event thread until the `configurationDone` request was called?

Do we miss any events that would have happened before this point?

Or maybe we don't subscribe to process events until we get the `configurationDone` request? Could that alleviate the need for this lock?

Reply:
I think that's not an option. Every process must have an event handler and IIUC (@jimingham should be able to confirm) if you don't handle the event, the process will never get out of the "launching" or "attaching" state.

Reply:
That is true, but it should only matter if someone actually cares about the process in the mean time. Provided we don't need to inspect the state, a sequence like this should be fine:
1. cause a process event to be emitted
2. start the listener thread
3. listener thread receives the event

But I don't know if that's the case here...

Another potential option: If we are sure the event is going to be emitted (and the listener thread is not running), maybe we can wait for the event (synchronously) on the thread which is processing the request.